check/check.go

```diff
@@ -120,31 +128,63 @@ func (p *Params) CheckForUpdate(url string, up *update.Update) (*Result, error)
 		return nil, err
 	}
 
-	resp, err := update.HTTPClient.Post(url, "application/json", bytes.NewReader(body))
+	req, err := http.NewRequest("POST", url, bytes.NewReader(body))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set("Content-Type", "application/json")
+
+	nonce := rand.Int63()
+	req.Header.Set("X-Message-Nonce", fmt.Sprintf("%d", nonce))
```

Comment:
Send a nonce to the update server, this nonce is supposed to be appended to the message before signing it.